homeassistant/components/lutron_caseta/__init__.py

```diff
@@ -39,23 +66,24 @@
 async def async_setup(hass, base_config):
     """Set up the Lutron component."""
 
-    bridge_configs = base_config[DOMAIN]
     hass.data.setdefault(DOMAIN, {})
 
-    for config in bridge_configs:
-        hass.async_create_task(
-            hass.config_entries.flow.async_init(
-                DOMAIN,
-                context={"source": config_entries.SOURCE_IMPORT},
-                # extract the config keys one-by-one just to be explicit
-                data={
-                    CONF_HOST: config[CONF_HOST],
-                    CONF_KEYFILE: config[CONF_KEYFILE],
-                    CONF_CERTFILE: config[CONF_CERTFILE],
-                    CONF_CA_CERTS: config[CONF_CA_CERTS],
-                },
+    if DOMAIN in base_config:
+        bridge_configs = base_config[DOMAIN]
+        for config in bridge_configs:
+            hass.async_create_task(
+                hass.config_entries.flow.async_init(
+                    DOMAIN,
+                    context={"source": config_entries.SOURCE_IMPORT},
+                    # extract the config keys one-by-one just to be explicit
+                    data={
+                        CONF_HOST: config[CONF_HOST],
```

Comment:
Can we take advantage of the fact that Lutron is HomeKit enabled or supports zeroconf and store its MAC address so we don't have to assign a static IP to the bridge in the router?

Reply:
In the underlying PR (https://github.com/home-assistant/core/pull/45136), if its setup via the UI, it will set the unique id and if its already found and the ip address changes it will update the config entry with the new host/ip here:

https://github.com/home-assistant/core/pull/45136/files#diff-a1f013e434276393714257521456ce864e12d014317713fc9eb9958a16ccb240R75